R/spatial_block_cv.R

```diff
@@ -177,7 +177,10 @@ systematic_block_cv <- function(data,
 generate_folds_from_blocks <- function(data, grid_blocks, v, n) {
   grid_blocks <- split_unnamed(grid_blocks, grid_blocks$fold)
 
-  indices <- row_ids_intersecting_fold_blocks(grid_blocks, data)
+  centroids <- sf::st_geometry(data)
```

Comment:
not sure how often you all use `sf`, so to clarify just in case: most geometry functions in `sf` throw a warning `<function> assumes attributes are constant over geometries of x` if you pass them an sf object with "attributes" (variables). `st_geometry` strips the variables and returns an `sfc` object, which won't throw the same warning.